moshi/src/test/java/com/squareup/moshi/ValueSourceEscapesTest.java

```diff
@@ -0,0 +1,68 @@
+package com.squareup.moshi;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.Okio;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public final class ValueSourceEscapesTest {
```

Comment:
This is adapted from https://github.com/infotech-group/moshi/commit/1d052e71e163b99a6f4b233f951edf9b71f646f8, but I wasn't able to get `escapedSingleQuote()` and `backslash1` working. Feedback welcome